### ISSUE TITLE
ClientWhiteList not working

### ISSUE THREAD

**Issue by odysseus1973** [stefanprodan/AspNetCoreRateLimit#22]
Hello!

I have the following settings

```
"IpRateLimiting": {
    "EnableEndpointRateLimiting": true,
    "StackBlockedRequests": false,
    "RealIpHeader": "X-Real-IP",
    "ClientIdHeader": "X-ClientId",
    "HttpStatusCode": 429,
    "IpWhitelist": [ "127.0.0.1", "::1/10", "192.168.0.0/24" ],
    "EndpointWhitelist": [ "get:/api/license", "*:/api/status" ],
    "ClientWhitelist": [ "test-client" ],
    "GeneralRules": [
      {
        "Endpoint": "*",
        "Period": "10s",
        "Limit": 5
      },
      {
        "Endpoint": "get:/v4/references/goods_links",
        "Period": "1h",
        "Limit": 1
      },
      {
        "Endpoint": "*",
        "Period": "15m",
        "Limit": 100
      }
      //{
      //  "Endpoint": "*",
      //  "Period": "12h",
      //  "Limit": 1000
      //}
      //},
      //{
      //  "Endpoint": "*",
      //  "Period": "7d",
      //  "Limit": 10000
      //}
    ]
  },
```

When I call endpoint get:/v4/references/goods_links  I always get rate limit even with header X-ClientId test-client

Thanks!

**Comment by markashton** [comment]
I also have the same problem

**Comment by odysseus1973** [comment]
I think that the project is abandoned, since it has not been updated for a long time.
I forked this project and corrected error with ClientWhitelist, also add initial wildcard support in paths

https://github.com/odysseus1973/AspNetCoreRateLimit

**Comment by markashton** [comment]
Thanks for the info. While debugging to verify my header was appearing ok, I did notice that under AspNetCore `X-ClientId` was appearing as `x-clientid`. So I've worked around the problem by defining `ClientIdHeader` as an all lowercase header name, but your solution (ignore the casing when grabbing the header) is obviously the better approach for the code.

**Comment by stefanprodan** [comment]
@odysseus1973 if you care to submit a PR I would gladly merge it. This year I've went through a career change and didn't got the time to maintain this project, sorry.

**Comment by stefanprodan** [comment]
Fixed by #25 thanks @odysseus1973
